### DISCUSSION TITLE
Reindex file or folder from web interface

### DISCUSSION THREAD

**Discussion by maximecurioni** [photoprism/photoprism#1614] [Ideas]
Having the ability to reindex a folder is very useful. The only downside is that it has to be performed on the server, using `docker-compose exec photoprism photoprism index <folder>`.  In my case, that means SSHing into the machine, finding the configuration directory, remembering the `reindex` command, etc.

If it's not too much work, it'd be useful to have the option to perform the reindexing of a single file or a single folder from the web UI directly.

**Comment by lastzero** [comment]
Any reason you don't use the existing dropdown for this? 🤔

![Screenshot_20211010_215142](https://user-images.githubusercontent.com/301686/136711154-d64a3758-5229-4334-a158-4fba576a7496.jpg)

**Reply by maximecurioni** [reply]
🤦‍♂️ 

Thanks @lastzero. Didn't know that was possible.

**Comment by maximecurioni** [comment]
My bad. This is already well-documented in the docs:
https://docs.photoprism.org/user-guide/library/originals/#indexing-originals

**Comment by maximecurioni** [comment]
@lastzero I have a follow-up question. I have a movie that exists, plays fine from the browser but shows up in the Hidden category. It's also missing its thumbnail.

Does a reindex trigger thumbnails to be regenerated?

(I want to troubleshoot this first before I file a possible Github issue)

**Reply by lastzero** [reply]
No, as we assume the file is broken when ffmpeg can't extract a still image for creating thumbs. That is typically the case, but there may be exceptions. Indexing and transcoding videos is a never ending story. Might as well stop doing photos and focus on videos 24/7. So many codecs and problems.

**Reply by maximecurioni** [reply]
Sounds good. Totally hear you that this can be rabbit hole, so I'll try to hand-fix it in the database.

Thanks!
